### PR TITLE
Tweeting branch PR

### PR DESCRIPTION
**Pull request by gokul-khoros**
The getUsertimeline, getHometimeline, and tweeting using update status are implemented in this branch(tweeting). Used Twitter4j which is added in maven dependencies.

### CODE REVIEW

File: src/main/java/RetriveTimeline.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import twitter4j.*;
+
+import java.util.List;
+
+public class RetriveTimeline {
+
+
+    public static void getTimeLine() throws TwitterException {
+        Twitter twitter = TwitterFactory.getSingleton();
+        Paging page = new Paging (1, 10); //page number, number per page
```
Comment (PrashanthRangarajan) [inline src/main/java/RetriveTimeline.java]: You can expose these values for user input and set the paging accordingly.

File: src/main/java/RetriveTimeline.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        List<Status> statuses = twitter.getUserTimeline("SonuSood",page);
```
Comment (PrashanthRangarajan) [inline src/main/java/RetriveTimeline.java]: What if I want my home feed to be displayed?

File: src/main/java/MainApplication.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import twitter4j.*;
+
+public class MainApplication {
+    public static void main(String[] args) throws TwitterException {
+        createTweet("My First tweet");
```
Comment (PrashanthRangarajan) [inline src/main/java/MainApplication.java]: Get user input for the tweet.

File: src/main/java/MainApplication.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        RetriveTimeline.getTimeLine();
+    }
+
+    public static void createTweet(String myTweet) throws TwitterException {
+        Twitter twitter = TwitterFactory.getSingleton();
+        Status status = twitter.updateStatus(myTweet);
+        System.out.println("Successfully updated the status : " + myTweet);
```
Comment (PrashanthRangarajan) [inline src/main/java/MainApplication.java:40]: Handle the exception scenario

File: src/main/java/RetriveTimeline.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import twitter4j.*;
+
+import java.util.List;
+
+public class RetriveTimeline {
+
```
Comment (PrashanthRangarajan) [inline src/main/java/RetriveTimeline.java]: Please remove unnecessary spaces

File: pom.xml [hunk 6]
```diff
@@ -7,6 +7,26 @@
     <groupId>org.example</groupId>
     <artifactId>TwittersAPiNew</artifactId>
     <version>1.0-SNAPSHOT</version>
```
Comment (PrashanthRangarajan) [inline pom.xml]: Please follow the maven naming conventions for artifactId: https://maven.apache.org/guides/mini/guide-naming-conventions.html

File: src/main/java/RetriveTimeline.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import twitter4j.*;
+
+import java.util.List;
+
+public class RetriveTimeline {
```
Comment (PrashanthRangarajan) [inline src/main/java/RetriveTimeline.java]: The class name shouldn't be a verb. Please change it to an appropriate noun.

File: src/main/resources/twitter4j.properties [hunk 8]
```diff
@@ -0,0 +1,6 @@
+
+#oauth.consumerKey = *****
```
Comment (PrashanthRangarajan) [inline src/main/resources/twitter4j.properties]: Why is this commented?

**Comment by PrashanthRangarajan** [comment]
The commit should be descriptive of what you are adding as part of the changeset. Also please add a detailed description to the PR for future reference.

File: src/main/java/RetriveTimeline.java (hunk 2 above)
Comment (gokul-khoros) [reply src/main/java/RetriveTimeline.java]: yea, then i will add a method for getHomeTimeline()

File: src/main/java/MainApplication.java (hunk 3 above)
Comment (gokul-khoros) [reply src/main/java/MainApplication.java]: both paging value and new tweet will be given by the user in next commit

File: src/main/java/RetriveTimeline.java (hunk 5 above)
Comment (gokul-khoros) [reply src/main/java/RetriveTimeline.java]: Sure

File: src/main/resources/twitter4j.properties (hunk 8 above)
Comment (gokul-khoros) [reply src/main/resources/twitter4j.properties]: Token and keys should not be disclosed so created these credentials with dummy values and commented it.
So while pushing the original credentials will be deleted.

Comment (PrashanthRangarajan) [reply src/main/resources/twitter4j.properties]: Then you can have these uncommented and add a single text comment informing the person who clones this branch to fill in the necessary oauth details in this property file

**Review by PrashanthRangarajan — APPROVED** [review]

**Pull request merged**